bzip2-sys/wasm_shim.rs

```diff
@@ -0,0 +1,55 @@
+use std::alloc::{alloc, dealloc, Layout};
+use std::os::raw::{c_int, c_void};
+
+#[no_mangle]
+pub extern "C" fn rust_bzip2_wasm_shim_malloc(size: usize) -> *mut c_void {
+    unsafe {
+        let layout = Layout::from_size_align_unchecked(size, 1);
```

Comment:
I think this will need to change, notably the alignment of 1. The default `malloc` alignment is at least the alignment of `usize` and I think it's actually twice that, although it'd be good to check what `malloc` guarantees.